staging/src/k8s.io/apiserver/pkg/version/types.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+// Info contains versioning information.
+// TODO: Add []string of api versions supported? It's still unclear
+// how we'll want to distribute that information.
+type Info struct {
```

Comment:
I don't think we can just move this here.  The version is also used by the client to produce output for things like `kubectl version` (the version of `kubectl`). I think we're going to end up with some real ugly to avoid bad dependencies.
 1. put it apimachinery.  apimachinery doesn't care about this, so its a bad choice.
 2. put it client-go, since the apiserver can depend on the client, but not the other way around.  Ugly, but workable.
 3. create a duplicate struct and copy in kubeapiserver via reflection or some kind of messed up scheme thing.